bolt/lib/Rewrite/DWARFRewriter.cpp

```diff
@@ -672,22 +672,36 @@ void DWARFRewriter::updateDWARFObjectAddressRanges(
     }
   }
 
+  Optional<AttrInfo> LowPCAttrInfo =
+      findAttributeInfo(DIE, dwarf::DW_AT_low_pc);
+  Optional<AttrInfo> HighPCAttrInfo =
+      findAttributeInfo(DIE, dwarf::DW_AT_high_pc);
   if (AbbreviationDecl->findAttributeIndex(dwarf::DW_AT_ranges)) {
     // Case 1: The object was already non-contiguous and had DW_AT_ranges.
     // In this case we simply need to update the value of DW_AT_ranges
     // and introduce DW_AT_GNU_ranges_base if required.
     Optional<AttrInfo> AttrVal = findAttributeInfo(DIE, dwarf::DW_AT_ranges);
-
     std::lock_guard<std::mutex> Lock(DebugInfoPatcherMutex);
     DebugInfoPatcher.addLE32Patch(
         AttrVal->Offset, DebugRangesOffset - DebugInfoPatcher.getRangeBase(),
         AttrVal->Size);
-    if (!RangesBase)
+
+    // high_pc attribute is unused, nullify if exists
+    if (HighPCAttrInfo) {
+      if (isHighPcFormEightBytes(HighPCAttrInfo->V.getForm()))
```

Comment:
@ayermolo Should I replace it with 
`TempDebugPatcher->addLE32Patch(HighPCOffset, HighPC, HighPCVal->Size);` ?

Reply:
Good question. The size parameter is needed if we are changing the form. Say going from 8 byte form to 4 byte form. It tells the code later how many bytes to skip in the original .debug_info section when it is copying chunks of data between patches. If nothing is specified its assumed size remains unchanged. So for addLE32Patch it remains 4 bytes. 

Since we are not changing form here we need to use corresponding addLE{64,32}Patch APIs I think.